examples/gg20_keygen_client.rs

```diff
@@ -0,0 +1,281 @@
+    )
+    .unwrap()
+    .into();
+
+    let client = Client::new();
+    let (party_num_int, uuid) = match signup(&client).unwrap() {
+        PartySignup { number, uuid } => (number, uuid),
+    };
+
+    let delay = time::Duration::from_millis(25);
+    let input_stage1 = KeyGenStage1Input {
+        index: (party_num_int - 1) as usize,
+    };
+    let res_stage1: KeyGenStage1Result = keygen_stage1(&input_stage1);
+
+    assert!(broadcast(
+        &client,
+        party_num_int,
+        "round1",
+        serde_json::to_string(&res_stage1.bc_com1_l).unwrap(),
+        uuid.clone()
+    )
+    .is_ok());
+
+    let round1_ans_vec = poll_for_broadcasts(
+        &client,
+        party_num_int,
+        params.share_count,
+        delay,
+        "round1",
+        uuid.clone(),
+    );
+    let mut bc1_vec = round1_ans_vec
+        .iter()
+        .map(|m| serde_json::from_str::<KeyGenBroadcastMessage1>(m).unwrap())
+        .collect::<Vec<_>>();
+
+    bc1_vec.insert(party_num_int as usize - 1, res_stage1.bc_com1_l);
+    assert!(broadcast(
+        &client,
+        party_num_int,
+        "round2",
+        serde_json::to_string(&res_stage1.decom1_l).unwrap(),
+        uuid.clone()
+    )
+    .is_ok());
+    let round2_ans_vec = poll_for_broadcasts(
+        &client,
+        party_num_int,
+        params.share_count,
+        delay,
+        "round2",
+        uuid.clone(),
+    );
+    let mut decom1_vec = round2_ans_vec
+        .iter()
+        .map(|m| serde_json::from_str::<KeyGenDecommitMessage1>(m).unwrap())
+        .collect::<Vec<_>>();
+    decom1_vec.insert(party_num_int as usize - 1, res_stage1.decom1_l);
+    let input_stage2 = KeyGenStage2Input {
+        index: (party_num_int - 1) as usize,
+        params_s: params.clone(),
+        party_keys_s: res_stage1.party_keys_l.clone(),
+        decom1_vec_s: decom1_vec.clone(),
+        bc1_vec_s: bc1_vec.clone(),
+    };
+    let res_stage2 = keygen_stage2(&input_stage2).expect("keygen stage 2 failed.");
+
+    let mut point_vec: Vec<GE> = Vec::new();
+    let mut enc_keys: Vec<BigInt> = Vec::new();
+    for i in 1..=params.share_count {
+        point_vec.push(decom1_vec[(i - 1) as usize].y_i);
+        if i != party_num_int {
+            enc_keys.push(
+                (decom1_vec[(i - 1) as usize].y_i.clone() * res_stage1.party_keys_l.u_i)
+                    .x_coor()
+                    .unwrap(),
+            );
+        }
+    }
+
+    let (head, tail) = point_vec.split_at(1);
+    let y_sum = tail.iter().fold(head[0], |acc, x| acc + x);
+
+    let mut j = 0;
+    for (k, i) in (1..=params.share_count).enumerate() {
+        if i != party_num_int {
+            // prepare encrypted ss for party i:
+            let key_i = BigInt::to_vec(&enc_keys[j]);
+            let plaintext = BigInt::to_vec(&res_stage2.secret_shares_s[k].to_big_int());
+            let aead_pack_i = aes_encrypt(&key_i, &plaintext);
+            assert!(sendp2p(
```

Comment:
In the identifiable abort variant those messages are broadcasted as well. Here I guess you use `sendp2p` for efficiency. I think that you should add a comment about that.

Reply:
done.